internal/cache/metadata/type_cache.go

```diff
@@ -64,13 +65,33 @@ type TypeCache interface {
 type cacheEntry struct {
 	expiry    time.Time
 	inodeType Type
+	// Copy of key string (internally only
+	// points to the original
+	// string data, so size overhead is a fixed 16 bytes, i.e. O(1),
+	// irrespective of actual key value.
+	// This is needed to calculate the
+	// accurate size of the type-cache entry on heap.
+	key string
 }
 
-// Size returns the size of cacheEntry.
-// It is currently set to dummy value 1 to avoid
-// the unnecessary  actual size calculation.
-func (ce cacheEntry) Size() uint64 {
-	return 1
+// Size returns the size of cacheEntry on RSS (Resident Set Size), which is
+// approximated as twice as calculated heap-size.
+// The calculated heap-size for each entry is
+// 80-bytes (24 for expiry, 8 for inodeType, 3*16 for 3 copies of
+// string structure for the key ) + length of the key itself,
+// for each entry (including negative ones).
+// This size is specific
+// to 64-bit linux machines, and might differ on other platforms.
+func (ce cacheEntry) Size() (size uint64) {
+	// First calculate size on heap.
+	// Additional 2*util.UnsafeSizeOf(&e.key) is to account for the 2 other copies of string
+	// struct stored in the cache map and in the cache linked-list.
+	size = uint64(util.UnsafeSizeOf(&ce) + 2*util.UnsafeSizeOf(&ce.key) + len(ce.key))
+
+	// Convert heap-size to rss (resident set size).
```

Comment:
would suggest you to create constant for heap to RSS conversion factor and use it everywhere.

Reply:
done.